clang/lib/AST/ASTContext.cpp

```diff
@@ -5148,6 +5153,14 @@ QualType ASTContext::getDependentBitIntType(bool IsUnsigned,
   return QualType(New, 0);
 }
 
+QualType ASTContext::getPredefinedSugarType(uint32_t KD,
+                                            QualType UnderlyingType) const {
+  auto *New = new (*this, alignof(PredefinedSugarType)) PredefinedSugarType(
```

Comment:
This should be uniqued like the other types.
If you drop the UnderlyingType, then this is really simple, as there are only three kinds and you can use a small array instead of a folding set node.